packages/schematics/angular/utility/dependencies.ts

```diff
@@ -0,0 +1,153 @@
+  packageName: string;
+  version: string;
+  overwrite?: boolean;
+}
+
+export function addPackageJsonDependencies(tree: Tree, dependencies: NodeDependency[]): void {
+    dependencies
+      .forEach(dependency => addPackageJsonDependency(tree, dependency));
+}
+
+export function addPackageJsonDependency(tree: Tree, dependency: NodeDependency): void {
+  const packageJsonAst = _readPackageJson(tree);
+  const dependencyPropName = _getDependencyPropName(dependency.type);
+  const depsNode = findPropertyInAstObject(packageJsonAst, dependencyPropName);
+  const recorder = tree.beginUpdate(pkgJsonPath);
+
+  if (!depsNode) {
+    // Haven't found the dependencies key, add it to the root of the package.json.
+    appendPropertyInAstObject(recorder, packageJsonAst, dependencyPropName, {
+      [dependency.packageName]: dependency.version,
+    });
+  } else if (depsNode.kind === 'object') {
+    // check if package already added
+    const depNode = findPropertyInAstObject(depsNode, dependency.packageName);
+
+    if (!depNode) {
+      // No build-angular package, add it.
+      appendPropertyInAstObject(
+        recorder,
+        depsNode,
+        dependency.packageName,
+        dependency.version,
+      );
+    } else if (dependency.overwrite) {
+      const { end, start } = depNode;
+      recorder.remove(start.offset, end.offset - start.offset);
+      recorder.insertRight(start.offset, JSON.stringify(dependency.version));
+    }
+  }
+
+  tree.commitUpdate(recorder);
+  _sortDependencies(tree, dependency);
+}
+
+export function getPackageJsonDependency(tree: Tree, packageName: string): NodeDependency | null {
+  const packageJson = _readPackageJson(tree);
+  let dep: NodeDependency | null = null;
+  [
+    NodeDependencyType.Default,
+    NodeDependencyType.Dev,
+    NodeDependencyType.Optional,
+    NodeDependencyType.Peer,
+  ].forEach(depType => {
+    const depPropName = _getDependencyPropName(depType);
+    if (dep !== null) {
+      return;
+    }
+    const depsNode = findPropertyInAstObject(packageJson, depPropName);
+    if (depsNode !== null && depsNode.kind === 'object') {
+      const depNode = findPropertyInAstObject(depsNode, packageName);
+      if (depNode !== null && depNode.kind === 'string') {
+        const version = depNode.value;
+        dep = {
+          type: depType,
+          packageName: packageName,
+          version: version,
+        };
+      }
+    }
+  });
+
+  return dep;
+}
+
+function _readPackageJson(tree: Tree): JsonAstObject {
+  const buffer = tree.read(pkgJsonPath);
+  if (buffer === null) {
+    throw new SchematicsException('Could not read package.json.');
+  }
+  const content = buffer.toString();
+
+  const packageJson = parseJsonAst(content, JsonParseMode.Strict);
+  if (packageJson.kind != 'object') {
+    throw new SchematicsException('Invalid package.json. Was expecting an object');
+  }
+
+  return packageJson;
+}
+
+function _getDependencyPropName(type: NodeDependencyType): string {
```

Comment:
Using a string enum for `NodeDependencyType` would allow the elimination of this function.